shell/platform/darwin/ios/ios_external_view_embedder.h

```diff
@@ -14,15 +14,15 @@ class IOSExternalViewEmbedder : public ExternalViewEmbedder {
  public:
   IOSExternalViewEmbedder(const std::shared_ptr<FlutterPlatformViewsController>&
                               platform_views_controller,
-                          std::shared_ptr<IOSContext> context);
+                          const IOSContext& context);
 
   // |ExternalViewEmbedder|
   virtual ~IOSExternalViewEmbedder() override;
 
  private:
   const std::shared_ptr<FlutterPlatformViewsController>&
       platform_views_controller_;
-  std::shared_ptr<IOSContext> ios_context_;
+  const IOSContext& ios_context_;
```

Comment:
What guarantee do you have that this reference will not be invalid when you go to use `ios_context_`?  I think this may be a show stopper for this PR =(

Reply:
Maybe I'm missing something here but my thought was`PlatformViewsIOS` controls the lifecycle of `IOSContext` and external_view_embedder will not out-live the platform view, so this reference will not be invalidated.

Reply:
You may be right.  If that's the case it's probably safer to overload the SubmitFrame method so that it takes in the IOSContext reference.  Then the ios_external_view_embedder doesn't have to hold onto the reference as an instance variable (which creates the possibility of a bug).

The concern is that you might have correct ownership today, but someone might rejigger stuff and it would invalidate that and create the possibility for a crash.

Reply:
Ugh, actually it might be a bit more difficult to pass it as a variable as I was thinking because of the interface it's implementing.  Give it a look, if it becomes too difficult we should at the very least write up the ownership in the docstring for the instance variable.  That wouldn't be as nice as eliminating the need for it though.